rules/java-net/src/main/resources/otagent/java-net.btm

```diff
@@ -0,0 +1,110 @@
+
+RULE java-net: Java HttpUrlConnection connect entry
+CLASS ^java.net.HttpURLConnection
+METHOD connect()
+HELPER io.opentracing.contrib.agent.OpenTracingHelper
+AT ENTRY
+IF !$0.connected && retrieveSpan($0) == null && !ignore($0)
+DO
+  associateSpan($0, getTracer().buildSpan($0.getRequestMethod())
```

Comment:
> use some annotations in the code

I think this is a better choice, and easier to use. This is also the purpose that our tracer provides a develop-toolkit besides auto-instrumentation(implicit) mechanism.